src/uv_tcp_listen.c

```diff
@@ -328,5 +413,12 @@ void UvTcpListenClose(struct UvTcp *t)
         uvTcpIncomingAbort(incoming);
     }
 
-    uv_close((struct uv_handle_s *)&t->listener, uvTcpListenCloseCbListener);
+    if (t->num_listeners) {
+        for (size_t i = 0; i < t->num_listeners; ++i) {
+            uv_close((struct uv_handle_s *)&t->listeners[i],
+                     uvTcpListenCloseCbListener);
+        }
+    } else {
```

Comment:
Shouldn't we remove this else branch and run an `assert(t->num_listeners > 0)` instead? I don't think there's a case where we have zero listeners (`UvListen` would have returned an error in the first place I think).

Reply:
If I remove the else block a lot of the integration unit tests (e.g. all tcp_connect, append, write, listen,...) will fail as the close_cb will not be invoked.

Reply:
Please let me know if the above suggestion to put the `UvTcpMaybeFireCloseCb(t)` call at the end of `uvTcpClose` instead solves this issue and the `else` branch can be removed.

Reply:
Moved the UvTcpMaybeFireCloseCb call at the end of the uvTcpClose instead now. And this way it works fine now. This way I could put back in the assertions into the UvTcpMaybeFireCloseCb as well.